rfcs/0130-Implementing Checks API in tc-github while preserving Statuses API

```diff
@@ -0,0 +1,44 @@
+# RFC 130 - Implementing Checks API
+* Comments: [#130](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/130)
+* Proposed by: @djmitche
+
+# Summary
+
+We want to implement Checks API without dropping the support os Statuses API. Statuses API would be the default. Our users
+will have the freedom to choose which API they want by indicating their choice in `.taskcluster.yml`.
+
+## Motivation
+
+So GitHub has come up with a new API to create status indicators in its UI, called Checks.
+Checks API isn't in any way compatible with the current Statuses API - status created with one cannot be updated with another;
+they are two totally separate sets of data. Calling both APIs would result in two sets of status indicators in the UI.
+Any third party tools and integrations that rely on Statuses API to work would stop working after we switched to Checks API.
+
+We want to implement Checks because it seems to be the future, and because it gives a much richer output and a whole bunch 
+of controls for our users.
+
+At the same time, we want to preserve compatibility with Statuses because some of our users use tools and integration that 
+rely on them.
+
+# Details
+
+Currently we create a task and a status in a single function that listens
+to github webhooks. The idea is to create a task in separate function that listens to webhooks;
+then listen to the pulse message (`task defined`? `artifact created`?) on a certain queue that signals about creation of a task.
+The messages would go to this or that queue based on some indication in `taskcluster.yml` - basically, "use Checks API" or the
+absence of "use Checks API". The presence of thet indication would result in adding a route (like a path to a queue?)
+to the task definition.
+
+So we would have two separate listeners for `task created` event, one would create statuses using Checks API, and another 
+would use Statuses API. Similarly, status updates would come in through separate channels as well, so we would have two 
+separate sets of status listeners as well - one for updating status indicators through Statuses API, and another through Checks.
```

Comment:
Couldn't there be one taskDefined listener which sets either status or defines a check, and another taskGroupResolved listener which sets status or marks a check as complete?  If there were one single listener for both, even duplicated taskGroupResolved messages could be handled, to set a different status each time.

If that were the case, other 'notification' things might become easier to implement, like maybe IRC notifications, if they aren't already sent?

Reply:
Statuses and Checks work a bit differently (for Checks, we need to save checkrun id into DB, and we don't have to concern ourselves with the group status, whereas Statuses is basically the group status), so I'd prefer two separate sets of functions for easy debugging and modification.

Reply:
But is the idea to have one check for the entire task graph or every single task.  Once we have dependencies (say, with a decision task),  we probably don't want every single task in the graph to have a check mark.  If we listen for the task group resolution, we can see what the outcome is and put either a check, X or neutral then provide the tools.tc.net link to give more detail as to why it resolved as it did

Reply:
Because there could be so many tasks that they become hard to figure out without the extra context afforded by the tools.tc.net ui.  A lot of projects at Mozilla which could benefit from taskcluster-github need to be cross-platform and have more than one test suite.  One example is the NSS project.

NSS runs builds and tests mac, arm64, linux32, linux64, windows32 and windows64.  Builds are separated from tests.  Each platform has between 1 and 6 variants (e.g. Debug, ASAN, Optimized).  Each platform has ~20-25 test suites.  All of these are multiplicative, so we have something between 120 and 900 tasks for each commit.  I know NSS isn't currently on tc-github, but it's the sort of project which would be really great to have running on Taskcluster through github.

If we had a single check "Taskcluster" which provides details view to the task group inspector, we can provide a lot more context for the various task outcomes for those which wish to see detail and a simple, easy to parse "Good", "Neutral", or "Failed" for those pushes.

Also thinking about the branch protection feature, if each task is a check itself, then the branch protection feature rules would need to be updated each time an individual task is added or removed from the .taskcluster.yml document, rather than saying "Taskcluster tests must pass".

Reply:
Hm, so we might want to have checks able to operate on the taskGroup level, as well as on the task level.

Reply:
I am now wondering if several check-suites can be created for a PR. We can group those things into check-suites maybe?

But yeah, that's a good point. I'll research some more how we can use checks for group status

Reply:
So it's impossible to have multiple check-suites, so we need to offer people a possibility to define that in `.taskcluster.yml`. I wonder if we will know, when the task resolution statuses start to com in, which decision task they belong to?